component/status.go

```diff
@@ -0,0 +1,106 @@
+// Copyright  The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package component
+
+import (
+	"errors"
+	"time"
+)
+
+type Status int
+
+const (
+	StatusOK Status = iota
+	StatusError
+)
+
+// StatusSource component that reports a status about itself.
+// The implementation of this interface must be comparable to be useful as a map key.
+type StatusSource interface {
+	ID() ID
+}
+
+type StatusEvent struct {
+	timestamp time.Time
+	status    Status
+	err       error
+}
+
+func (ev *StatusEvent) Status() Status {
+	return ev.status
+}
+
+// Timestamp returns the time of the event.
+func (ev *StatusEvent) Timestamp() time.Time {
+	return ev.timestamp
+}
```

Comment:
Do you have use-case for this? If yes, do we need to allow this to be configured or we always do Time.Now in the `NewStatusEvent`.

Reply:
No, I think this can be deleted for now.

Reply:
+1

Reply:
I deleted timestamp from everywhere.